src/main/java/com/codingame/codemachine/compiler/java/CodinGameJavaCompiler.java

```diff
@@ -1,126 +1,105 @@
-                        log.setColumn(diagnostic.getColumnNumber());
-                        log.setMessage(diagnostic.getMessage(null));
-                        log.setKind(diagnostic.getKind() == Kind.ERROR ? CompilationLogKind.ERROR : CompilationLogKind.WARNING);
-                        logs.add(log);
+                    case MANDATORY_WARNING:
+                        type = "WARNING";
                         break;
-                    default:
-                        // nothing
+                    case NOTE:
+                        type = "INFO";
+                        break;
+                    case OTHER:
+                        continue;
+                    }
+                    
+                    System.err.println(String.format("%s:%d: %s: %s\n%s\n%s^", 
+                            diagnostic.getSource().getName(),
+                            diagnostic.getLineNumber(),
+                            diagnostic.getKind().name().toLowerCase(),
+                            diagnostic.getMessage(null),
+                            line,
+                            String.join("", Collections.nCopies((int)diagnostic.getColumnNumber() - 1, " "))
+                            ));
+                    
+                    System.out.println(String.format("CG> annotate --type \"%s\" --file \"%s\" --position \"%s\" --message \"%s\"",
+                            type,
+                            diagnostic.getSource().getName(),
+                            diagnostic.getLineNumber()+":"+diagnostic.getColumnNumber(),
+                            diagnostic.getMessage(null).replaceAll("\"","\\\"")
+                            ));
                 }
+                resultCode = success ? 0 : 1;
             }
-            fileManager.close();
-
-            result.setSuccess(success);
-            result.setLogs(logs);
-            resultCode = success ? 0 : 1;
-        }
-        else {
-            result.setSuccess(false);
-            CompilationLogDto log = new CompilationLogDto();
-            log.setKind(CompilationLogKind.ERROR);
-            log.setMessage("no source file");
-            result.setLogs(singletonList(log));
-            resultCode = 2;
-        }
-
-        String resultOutput = System.getProperty("codingame.compiler.output", DEFAULT_OUTPUT);
-        String resultStr = new Gson().toJson(result);
-        if (DEFAULT_OUTPUT.equals(resultOutput)) {
-            realOut.println(resultStr);
-        }
-        else {
-            try {
-                FileUtils.writeStringToFile(new File(resultOutput), resultStr);
-            }
-            catch (IOException e) {
-                realErr.println(e.getMessage());
-                System.exit(3);
+            else {
+                System.err.println("no source file");
+                resultCode = 2;
             }
+            System.out.println("CG> redirect-streams -r -i \"err\" \"compilation\"");
```

Comment:
I would prefer `--reset` instead of `-r`. I though the reset option stands for "Cancel all the redirections". In this case, is the rest of the command mandatory?